nuxeo-platform-3d-core/src/main/java/org/nuxeo/ecm/platform/threed/listener/ThreeDBatchGenerationListener.java

```diff
@@ -55,17 +54,9 @@ public void handleEvent(Event event) {
         if (doc.hasFacet(THREED_FACET) && !doc.isProxy()) {
             ThreeDBatchUpdateWork work = new ThreeDBatchUpdateWork(doc.getRepositoryName(), doc.getId());
             WorkManager manager = Framework.getService(WorkManager.class);
-
-            ThreeDBatchUpdateWork running = (ThreeDBatchUpdateWork) manager.find(work.getId(), Work.State.RUNNING);
-            ThreeDBatchUpdateWork scheduled = (ThreeDBatchUpdateWork) manager.find(work.getId(), Work.State.SCHEDULED);
-            if (running != null) {
-                running.suspended();
-                running.setStatus("Suspended");
-            } else if (scheduled != null) {
-                scheduled.suspended();
-                scheduled.setStatus("Suspended");
-            }
-
+            manager.schedule(work, WorkManager.Scheduling.CANCEL_SCHEDULED, true);
+            // this is necessary since StreamWorkManager implementation does not automatically
+            // reschedule the work given with CANCEL_SCHEDULED
```

Comment:
The initial intend afaiu is to cancel an exising work then resubmit a new one,
to do this first the ThreeDBatchUpdateWork must be declared not being idempotent because 2 works with the same id are not equivalent.
Also canceling a scheduled work is supported when storeState is enabled, but the canceled state is not reset once it is taken in account, the result is that the new submitted work will be canceled.
If no unit test warn about this this worker need an heavier refactoring that deserves its own ticket.